cmd/k8s-operator/ports.go

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package main
```

Comment:
Nit: can we name this file something more specific to the NodePort Services? We have other bits of fairly complicated port wrangling in other places (see egress services logic, for example)